src/statick_tooling/plugins/tool/hadolint_tool_plugin.py

```diff
@@ -0,0 +1,187 @@
+        else:
+            return None
+
+        for output in total_output:
+            logging.debug("%s", output)
+
+        with open(self.get_name() + ".log", "w") as fid:
+            for output in total_output:
+                fid.write(output)
+
+        issues: List[Issue] = self.parse_output(total_output)
+        return issues
+
+    # pylint: enable=too-many-locals
+
+    def scan_local_binary(
+        self, tool_bin: str, flags: List[str], files: List[str]
+    ) -> Optional[str]:
+        """Use locally installed hadolint binary to scan."""
+        try:
+            exe = [tool_bin] + flags
+            exe.extend(files)
+            output = subprocess.check_output(
+                exe, stderr=subprocess.STDOUT, universal_newlines=True
+            )
+            return output
+
+        except subprocess.CalledProcessError as ex:
+            logging.warning("%s failed! Returncode = %d", tool_bin, ex.returncode)
+            logging.warning("%s exception: %s", self.get_name(), ex.output)
+            return None
+
+        except OSError as ex:
+            logging.warning("Couldn't find %s! (%s)", tool_bin, ex)
+            return None
+
+    def scan_docker(
+        self, tool_bin: str, flags: List[str], files: List[str], config_file_path: str
+    ) -> Optional[str]:
+        """Use hadolint docker image to scan."""
+        try:
+            json_dict = []
+            for src in files:
+                exe = [
+                    "docker",
+                    "run",
+                    "--rm",
+                    "-i",
+                    "-v",
+                    config_file_path + ":/.config/hadolint.yaml",
+                    "-v",
+                    src + ":/Dockerfile",
+                    "hadolint/hadolint",
+                    "hadolint",
+                ]
+                exe.extend(flags)
+                exe.append("Dockerfile")
+                output = subprocess.check_output(
+                    exe, stderr=subprocess.STDOUT, universal_newlines=True
+                )
+                output = output.replace('"file":"Dockerfile"', '"file":"' + src + '"')
+                file_dict = json.loads(output)
+                for issue in file_dict:
+                    json_dict.append(issue)
+            return json.dumps(json_dict)
+
+        except subprocess.CalledProcessError as ex:
+            logging.warning("%s failed! Returncode = %d", tool_bin, ex.returncode)
+            logging.warning("%s exception: %s", self.get_name(), ex.output)
+            return None
+
+        except OSError as ex:
+            logging.warning("Couldn't find %s! (%s)", tool_bin, ex)
+            return None
+
+    def parse_output(self, total_output: List[str]) -> List[Issue]:
+        """Parse tool output and report issues."""
+        issues: List[Issue] = []
+
+        # pylint: disable=too-many-nested-blocks
+        for output in total_output:
+            for line in output.split("\n"):
```

Comment:
```suggestion
            for line in output.splitlines():
```

Using `splitlines` handles universal line separators and can be more portable.

https://docs.python.org/3/library/stdtypes.html#str.splitlines